### ISSUE TITLE
Update FlinkCatalog to use Namespace for baseNamespace field for cleaner output

### ISSUE THREAD

**Issue by kbendick** [apache/iceberg#1541]
The FlinkCatalog currently has a `String[] baseNamespace` field. However, when adding in names to the parameterized tests in this PR: https://github.com/apache/iceberg/pull/1539, I noticed that the output for the `baseNamespaces` parameters came out as `(new String[]{"a", "b"}).toString()`.

In order to make the tests more human readable, we're going to accept that a baseNamespace is simply a prefix of a `Namespace` (especially considering that there's not really a limit on the number of dots in a namespace).

I'm going to refactor FlinkCatalog to use `Namespace` for that field so that we can take advantage of its preexisting static constructor `Namespace.from(String...)` as well as its proper `.toString` implementation.

This will make test output much more readable so we can see what values we're actually testing with just by running the tests, as well as potentially making stack traces that involve these fields easier to read.

**Comment by kbendick** [comment]
Currently, the test output for the parameterized name looks like the following when the tests are run (for reference)
```
testReplacePartitions[catalogName=testhadoop, baseNamespace=[Ljava.lang.String;36c1c7d9], format=PARQUET, isStreaming=false]
```

**Comment by rdblue** [comment]
Fixed by #1984.
